src/createMatcher.ts

```diff
@@ -2,14 +2,15 @@
 
 import path from 'path'
 
-import multimatch from 'multimatch'
+import picomatch from 'picomatch'
 
 export default (pattern: string[] | string | undefined) => {
   // Match everything if no pattern was given
   if (typeof pattern !== 'string' && !Array.isArray(pattern)) {
     return () => true
   }
   const patterns = Array.isArray(pattern) ? pattern : [pattern]
-  return (file: string) =>
-    multimatch(path.normalize(file), patterns, { dot: true }).length > 0
+
+  const isMatch = picomatch(patterns, { dot: true })
+  return (file: string) => isMatch(path.normalize(file))
```

Comment:
Using `picomatch` here also helped hoisting, which prevents `patterns` from being evaluated multiple times.

Reply:
Will any issues be fixed by this change?

https://github.com/prettier/pretty-quick/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+pattern

Reply:
Doesn't know. `multimatch` is powered by the `minimatch` core. `picomatch` itself is a core that powers `nanomatch` and `micromatch`. And they do have different behaviors.

Though the existing test cases are passed, I don't know if those issues are solved. I will create another PR adding test cases covering those issues.